### ISSUE TITLE
CentOS 7 compile issue

### ISSUE THREAD

**Issue by KristijanL** [threerings/openvpn-auth-ldap#51]
Hi,

i try to compile openvpn-auth-ldap on CentOS 7.1 without success.
issue:

[root@vpn openvpn-auth-ldap]# ./configure --prefix=/opt/openvpn-auth-ldap --with-openvpn=/opt/SW/openvpn-2.3.8
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for clang... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang accepts -g... yes
checking for clang option to accept ISO C89... none needed
checking for clang option to accept ISO C99... none needed
checking for clang... clang
checking whether we are using the GNU Objective C compiler... yes
checking whether clang accepts -g... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether ln -s works... yes
checking whether make sets $(MAKE)... yes
checking for ranlib... ranlib
checking for ar... ar
checking for re2c... /usr/local/bin/re2c
checking for doxygen... no
checking for dot... no
checking how to run the C preprocessor... clang -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for the pthreads library -lpthreads... no
checking whether pthreads work without any flags... no
checking whether pthreads work with -Kthread... no
checking whether pthreads work with -kthread... no
checking for the pthreads library -llthread... no
checking whether pthreads work with -pthread... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking if more special flags are required for pthreads... no
checking for BSD pf(4) support... no
configure: WARNING: pf(4) table support will not be included.
checking for strlcpy... no
checking for openldap... yes
checking for openssl... yes
checking for NSStringFromSelector in Foundation framework... no
checking for openvpn-plugin.h... yes
checking how to run the Objective C preprocessor... clang -E
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking objc/objc.h usability... no
checking objc/objc.h presence... no
checking for objc/objc.h... no
configure: error: Can't locate Objective C runtime headers
###

yum install gcc-objc
Package gcc-objc-4.8.3-9.el7.x86_64 already installed and latest version
Nothing to do

[root@vpn openvpn-auth-ldap]# find / | grep objc.h
/usr/lib/gcc/x86_64-redhat-linux/4.8.2/include/objc/objc.h

am i missing anything?

Best Regards,
Kristijan

**Comment by tatref** [comment]
Same issue here. I couldn't get openvpn-auth-ldap to compile on any recent centos.
I would be great to have a list of yum dependecies to install to order to get reproduceable builds.

**Comment by optiwiper** [comment]
Hi, same problem with ArchLinux.

**Comment by romanlaube** [comment]
Hi,
I have similar problem. In "config.log" I see "fatal error: objc/objc-api.h: No such file or directory". I have also found that file "objc-api.h" is no longer available in GCC (since 4.7.0) - see https://gcc.gnu.org/onlinedocs/gcc/Traditional-GNU-Objective-C-runtime-API.html - traditional vs modern API...
Roman Laube

**Comment by HidekiAI** [comment]
I do not think it has anything to do with the plugin, but rather/more-so to do with your gcc setup.  At least for me (on Gentoo), I've cheated and did:
`# ln -sv /usr/lib64/gcc/x86_64-pc-linux-gnu/4.9.3/include/objc /usr/include/objc`
to get around the issue of not being able to find 'objc.h' file during config and was able to build it.
